File: python/Transformed_Array_3379.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        results = [None] * len(nums)
        for i in range(len(nums)):
            results[i] = nums[(i + nums[i] + 100 * len(nums)) % len(nums)]
        return results

File: cpp/Transformed_Array_3379.cpp
class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {
        vector <int> result; result.reserve(nums.size());
        for (int i = 0; i < nums.size(); i++) {
            result.push_back(nums[(i + nums[i] + 100 * nums.size()) % nums.size()]);
        }
        return result;
    }
};
